clients/common/hipblas_arguments.cpp

```diff
@@ -16,6 +16,7 @@ std::ostream& operator<<(std::ostream& os, std::pair<T1, T2> p)
 {
     os << p.first << ": ";
     os << p.second;
+    return os;
```

Comment:
You can turn this sort of omission into a compiler error with the flag `-Werror=return-type`. It has basically no false-positives, but I suppose enabling *any* warnings as errors at all is probably a bad idea for a default in an OSS library. We could probably add an option that the CI opts into, though.

Reply:
Yes we don't do that for many warnings, usually fixing any warnings is the best policy.   Leave it to @daineAMD  to decide if he wants to adjust hipBLAS I am fine either way.